HPCC-20631 Set session id context

mayx opened the pull request:
- Set the session id context of an ssl context to be "hpccsystems", a
  value unique to our applicaiton.

Signed-off-by: mayx <yanrui.ma@lexisnexisrisk.com>

<!-- Thank you for submitting a pull request to the HPCC project

 PLEASE READ the following before proceeding.

 This project only accepts pull requests related to open JIRA issues.
 If suggesting a new feature or change, please discuss it in a JIRA issue first.
 If fixing a bug, there should be an issue describing it with steps to reproduce.
 The title line of the pull request (and of each commit within it) should refer to the
 associated issue using the format:

 HPCC-nnnnn Short description of issue

 This will allow the Jira ticket to be automatically updated to refer to this pull request,
 and will ensure that the automatically-generated changelog is properly formatted.
 Where a pull request contains a single commit the pull request title will be set automatically,
 assuming that the commit has followed the proper guidelines.

 Please go over all the following points, and put an `x` in all the boxes that apply. You may find
 it easier to press the 'Create' button first then click on the checkboxes to edit the comment.
-->

## Type of change:
- [x] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [x] All new and existing tests passed.
  - [x] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [x] Scalability
  - [x] Performance
  - [x] Security
  - [x] Thread-safety
  - [x] Premature optimization
  - [x] Existing deployed queries will not be broken
  - [x] This change fixes the problem, not just the symptom
  - [x] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [x] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Testing:
<!-- Please describe how this change has been tested.-->

<!-- Thank you for taking the time to submit this pull request and to answer all of the above-->

HPCCSmoketest commented:
This PR (11766) is in the 1/1 position of the queue. Estimated start time is after ~0.5 hour(s)

HPCCSmoketest commented:
2/2. Process of PR-11766, label: HPCC-20631-VerifyCert-6.4 starts now. Estimated completion time is ~ 0.5 hour(s)

HPCCSmoketest commented:
Automated Smoketest: :white_check_mark:
OS: centos 7.4.1708 (Linux 3.10.0-327.28.3.el7.x86_64)
Sha: 27107dd9e9eea58a71b00dff75073b11ead8782d
Build: success
Install hpccsystems-platform-community_6.4.28-closedown0.el7.x86_64.rpm
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | 
|---|---|---|---|---|---|
|unittest |97 |97 |0 |0 |0 |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |
|wutoolTest(Cassandra) |19 |19 |0 |0 |0 |



Regression test result:

| phase | total | pass | fail | 
|---|---|---|---|
|setup (hthor) |11 |11 |0 |
|setup (thor) |11 |11 |0 |
|setup (roxie) |11 |11 |0 |
|test (hthor) |736 |736 |0 |
|test (thor) |631 |631 |0 |
|test (roxie) |765 |765 |0 |


HPCC Stop: OK
HPCC Uninstall: OK
Time stats:

| Prep time | Build time | Package time | Install time | Start time | Test time | Stop time | Summary | 
|---|---|---|---|---|---|---|---|
|   27 sec (00:00:27) |  142 sec (00:02:22) |   66 sec (00:01:06) |   18 sec (00:00:18) |   65 sec (00:01:05) | 1266 sec (00:21:06) |   40 sec (00:00:40) | 1624 sec (00:27:04) |

mayx committed:
HPCC-20631 Set session id context

- Set the session id context of an ssl context to be "hpccsystems", a
  value unique to our applicaiton.

Signed-off-by: mayx <yanrui.ma@lexisnexisrisk.com>

HPCCSmoketest commented:
This PR (11766) is in the 2/2 position of the queue. Estimated start time is after ~1.0 hour(s)

HPCCSmoketest commented:
This PR (11766) is in the 1/1 position of the queue. Estimated start time is after ~0.5 hour(s)

HPCCSmoketest commented:
3/3. Process of PR-11766, label: HPCC-20631-VerifyCert-6.4 starts now. Estimated completion time is ~ 0.5 hour(s)

HPCCSmoketest commented:
Automated Smoketest: :white_check_mark:
OS: centos 7.4.1708 (Linux 3.10.0-327.28.3.el7.x86_64)
Sha: ca8c7d4fc75684677ef68ccadd7fa925eb8b1329
Build: success
Install hpccsystems-platform-community_6.4.28-closedown0.el7.x86_64.rpm
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | 
|---|---|---|---|---|---|
|unittest |97 |97 |0 |0 |0 |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |
|wutoolTest(Cassandra) |19 |19 |0 |0 |0 |



Regression test result:

| phase | total | pass | fail | 
|---|---|---|---|
|setup (hthor) |11 |11 |0 |
|setup (thor) |11 |11 |0 |
|setup (roxie) |11 |11 |0 |
|test (hthor) |736 |736 |0 |
|test (thor) |631 |631 |0 |
|test (roxie) |765 |765 |0 |


HPCC Stop: OK
HPCC Uninstall: OK
Time stats:

| Prep time | Build time | Package time | Install time | Start time | Test time | Stop time | Summary | 
|---|---|---|---|---|---|---|---|
|   19 sec (00:00:19) |  149 sec (00:02:29) |   67 sec (00:01:07) |   16 sec (00:00:16) |   63 sec (00:01:03) | 1276 sec (00:21:16) |   39 sec (00:00:39) | 1629 sec (00:27:09) |
